File: MealMaster/MealMaster.BLL/DTOs/Request/Recipe/RecipeFilterDto.cs
namespace MealMaster.BLL.DTOs.Request.Recipe;

public class RecipeFilterDto
{
    public Guid UserId { get; set; }
    public string SearchTerm { get; set; }
    public int? MinCalories { get; set; } 
    public int? MaxCalories { get; set; } 
    public int? MinPreparationTime { get; set; } 
    public int? MaxPreparationTime { get; set; } 
    public Guid? CuisineTypeId { get; set; }
    public Guid? RestrictionId { get; set; }
}

File: Client/meal-master/src/app/models/recipe-filter.dto.ts
export interface RecipeFilterDto {
  userId: string;
  searchTerm: string;
  minCalories?: number | null;
  maxCalories?: number | null;
  minPreparationTime?: number | null;
  maxPreparationTime?: number | null;
  cuisineTypeId?: string | null;
  restrictionId?: string | null;
}
